### PR TITLE
Added nodeJS to the Dockerfile

### PR DESCRIPTION
**Pull request by RamuKammagani-osi**
Hi @edgibbs i have just updated the Dockerfile Please review the changes.

### CODE REVIEW

**Comment by RamuKammagani-osi** [comment]
Closes #20

File: Dockerfile [hunk 1]
```diff
@@ -1,34 +1,39 @@
-FROM ruby:2.3.3 
+FROM ruby:2.3.3
```
Comment (eckdanny-osi) [inline Dockerfile]: Lots of trailing whitespace in the diff. This is the way it should be tho. 👍

Comment (edgibbs) [reply Dockerfile]: We should be ruby:2.4.2 to match the project

File: Dockerfile [hunk 2]
```diff
@@ -1,34 +1,39 @@
...
-# will be cached unless changes to one of those two files 
+# Copy the Gemfile as well as the Gemfile.lock and install
+# the RubyGems. This is a separate step so the dependencies
+# will be cached unless changes to one of those two files
 # are made.
 
-COPY Gemfile Gemfile.lock ./ 
+COPY Gemfile Gemfile.lock ./
 RUN gem install bundler && bundle install --jobs 20 --retry 5
+RUN apt-get update
```
Comment (eckdanny-osi) [inline Dockerfile]: Its really uncommon to see multiple `apt-get` commands in a `Dockerfile` b/c each one adds a layer to the image's file system. I'd suggest consolidating all the _install-related_ bits and chaining with `&&`.

I can clean up later and introduce some other [`Dockerfile` Best Practices](https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/) in a follow up PR.

**Comment by eckdanny-osi** [comment]
Added improvements (and correct `rails` version) in #21. We can close #20 _in-favor-of_ or pull in #21 at a later time.

@RamuKammagani-osi normally I'd have paired up and pushed directly to your PR branch (with your permission), but its off hours and I don't want to impact your PR negatively. #21 branches off of 7aaadab771c4842b8d2ba0c5366d01b6821522a0. Hoping this approach affords us more flexibility.

**Comment by eckdanny-osi** [comment]
closing in favor of #21

**Pull request closed without merging**